forch/endpoint_handler.py

```diff
@@ -13,54 +13,100 @@
 import logging
 import subprocess
 import time
+import threading
+import yaml
 import grpc
 
 from forch.utils import get_logger
-
-try:
-    import daq.proto.session_server_pb2_grpc as server_grpc
-    from daq.proto.session_server_pb2 import SessionParams, SessionProgress
-    SessionServerServicer = server_grpc.SessionServerServicer
-    PROTO_LOADED = True
-except ImportError:
-    PROTO_LOADED = False
-
-    class SessionServerServicer:
-        """Dummy class for weak import"""
+import forch.proto.endpoint_server_pb2_grpc as server_grpc
+from forch.proto.endpoint_server_pb2 import Endpoint
+from forch.proto.shared_constants_pb2 import Empty
 
 DEFAULT_SERVER_PORT = 50111
 DEFAULT_BIND_ADDRESS = '0.0.0.0'
 DEFAULT_VXLAN_PORT = 4789
-DEFAULT_VXLAN_VNI = 0
-
+BASE_T1SW_PORT = 29
 VXLAN_CMD_FMT = 'ip link add %s type vxlan id %s remote %s dstport %s srcport %s %s nolearning'
-
+T1_SW1 = 'nz-kiwi-t1sw1'
 CONNECT_TIMEOUT_SEC = 60
 
 
 class EndpointHandler:
     """Class to handle endpoint updates"""
 
-    def __init__(self, target_ip):
+    def __init__(self, target_ip, structural_config_file):
         self._logger = get_logger('endpproxy')
+        self._lock = threading.RLock()
+        self._mac_t1sw1_port = {}
```

Comment:
Don't encode "t1sw1" here (variable name) -- at the very least, it's requiring somebody who's trying to understand this code to then understand the switch hierarchy. I'd just replace all instances of t1sw1 with "egress" or something more semantically meaningful.

Reply:
There is already port 28 named egress that goes to corp. Is there another named that can be used for these ports?